### PR TITLE
Slurm affinity checks with hybrid_check and check_gpu

### PR DESCRIPTION
**Pull request by mszpindler**
Simple test for slurm thread binding using `hybrid_check` tool. Single task per node check. Loosely based on CSCS version.

### CODE REVIEW

File: checks/slurm/affinity_check.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        self.executable_opts = ['-r']
+
+    @run_after('init')
+    def set_threading(self):
+        if self.multithread:
+            self.use_multithreading = True
+            self.variables['OMP_PLACES'] = 'threads' 
+        else:
+            self.use_multithreading = False
+            self.variables['OMP_PLACES'] = 'cores'
```
Comment (rsarm) [inline checks/slurm/affinity_check.py:65]: Maybe 
```suggestion
            self.variables['OMP_PLACES'] = self.omp_bind 
```
?

File: checks/slurm/affinity_check.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        else:
+            self.use_multithreading = False
+            self.variables['OMP_PLACES'] = 'cores' 
+ 
+
+    @run_before('run')
+    def set_omp_vars(self):
+        self.variables['OMP_PROC_BIND'] = 'close'
+            #'OMP_PLACES': 'cores',
+            #'OMP_PROC_BIND': 'close',
```
Comment (rsarm) [inline checks/slurm/affinity_check.py:72]: Maybe delete these comments?

File: checks/slurm/affinity_check.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    @run_before('run')
+    def set_omp_vars(self):
+        self.variables = {
+            'OMP_PLACES': self.omp_bind,
+            'OMP_PROC_BIND': self.omp_proc_bind,
+        }
+
+@rfm.simple_test
+class HybridCheck(AffinityOpenMPBase):
```
Comment (rsarm) [inline checks/slurm/affinity_check.py:47]: Why not a single class instead of the three classes?

**Comment by mszpindler** [comment]
Rewritten to support GPU binding and refer to the [https://github.com/Lumi-supercomputer/lumi-userguide](LUMI Docs) closely. Currently requires specific module set with Reframe: `--map-module LUMI:LUMI/22.12 --map-module lumi-CPEtools:lumi-CPEtools/1.1-cpeGNU-22.12`.

**Pull request merged**